### ISSUE TITLE
Meter overhaul

### ISSUE THREAD

**Issue by akleja** [tenacityteam/tenacity-legacy#399]
- [x] I have read the specified guidelines for issues

**Is your feature request related to a problem? Please describe.**
I find the meters virtually useless. I propose some improvements:

**Describe the solution you'd like**

* Read out values for Peak and RMS displayed on the meter.
* Flat color zones instead of gradient, with configurable division between low signal and nominal (default at either -18 or -20 dBFS.
* Oversampled "True Peak" metering option
* User adjustable response time of RMS meter
* LUFS as selectable alternative to RMS
* Make vertical meters a viable option by using a vertical toolbar when vertical meter is selected.
* Add docking areas to the side of the main window for vertical toolbars.

Some initial tinkering to add readouts (not finished): 
![meters](https://user-images.githubusercontent.com/49785979/127168308-8b7a4e4c-bcae-487c-a166-5b85b08929ed.png)

**Comment by Reaper10** [comment]
This may all so  help here #212

**Comment by petersampsonaudacity** [comment]
@akleja  Do you realize that flat colour meters are already available:

1) click on the meter icon,
2) select Options,
3) in Meter Style  choose "RMS".

Each meter must be set independently.

**Comment by poyribn** [comment]
> Do you realize that flat colour meters are already available

Most users don't change default settings,
so it's a great idea to make
Flat Color Zones
the default, since it's much more informationally clear than a gradient.

**Comment by akleja** [comment]
@petersampsonaudacity Yeah I know but that's not what I described. I meant that the meter should be divided in static flat colored zones with a division between green and yellow at nominal levels when in dBFS mode.
I'll draw up some illustrations later to be more clear about what I mean.

**Comment by petersampsonaudacity** [comment]
> .... I meant that the meter should be divided in static flat colored zones with a division between green and yellow at nominal levels when in dBFS mode.
> I'll draw up some illustrations later to be more clear about what I mean.

Like the ones in DarkAudacity?

But thet'd be better with numbers ...

![image](https://user-images.githubusercontent.com/50205984/127344594-715e9ccd-c873-42f0-98b5-3dae6b6f2b99.png)

**Comment by akleja** [comment]
Yeah precisely, and perhaps the interlacing should be optional

**Comment by iMartyn** [comment]
I really like the idea of optionally adding a current and peak db readout.  I also agree that they should always be flat colours by default, ideally Green-Yellow-Red not "some nice colours fitting the theme" (fitting the theme should be an optional setting IMO with the standard meter functionality being unaffected by this).
(This comment contains a distillation of some of my comments in https://github.com/tenacityteam/tenacity/discussions/448 )

**Comment by akleja** [comment]
Yeah I agree with sticking with standard colors and making theme colors optional.
I prefer the green and red to be darker than the yellow though to have colorblind people in mind. I think the optional theme based setting should rely on fairly distinct levels of brightness between the sections so it can be even more friendly to colorblind people.

Here's my current idea for the peak meter:
![meters](https://user-images.githubusercontent.com/49785979/129057913-dc71e19f-f4e6-4dda-ae8f-5d23168ea692.png)
( this is a drawing, not a screenshot )
The yellow is pushed slightly to the warmer side, and the green and red slightly to the cooler. I think the colors could be tweaked further. And I'm uncertain if there should be readouts for both channels.

**Comment by poyribn** [comment]
There Should be readouts for the left and right channels separately.
Every audio program does this for a good reason:

Audio playing in only one of both audio channels can be an embarrasing mistake, and having the meters differentiate between both stereo channels makes it easy to spot, which is important.

**Comment by iMartyn** [comment]
I think the yellow area should not change to orange, that makes things very confusing.  e.g. in the 3rd image, the right channel should either have 3 colours (green, yellow, orange) or I'm misunderstanding what you're trying to convey here? to me, you're using the whole area above 18 as changeable, that's highly unusual, if it's a yellow area it should stay yellow and have a clear line where it goes into the next colour imo.

**Comment by akleja** [comment]
@iMartyn Yeah you understood correctly. I'm used to this meter style, but if it's too non-standard I'll conform. I prefer having the yellow area changable, but that might be an effect of my impaired color vision. I could never spot the color changing to orange otherwise since the area is too small. But if that's too strange I'll happily stick with standard color divisions.


@Sosasees I agree at least when it comes to master meters, as there that extra level of detail is often needed. But on track meters it's not the norm. Ardour, Logic and Reaper to name a few only shows the one highest value in mixer and tcp meters. This makes comparing/balancing tracks easier.I think the scenario of just having audio in one channel would be obvious from the meters themselves.

**Comment by akleja** [comment]
So here's an updated concept with more conventional meter coloring, and a bigger dual readout master meter

![meters3](https://user-images.githubusercontent.com/49785979/129072985-c9139cf6-a7bd-4e59-a6b1-a694516e54df.png)


One thing I'm far from certain about is how the colors zones should be on a RMS/LUFS meter. I've seen many different approaches. Some use colors to distinguish between long term/short term when metering in LUFS.

**Comment by iMartyn** [comment]
That's so much clearer to me, because you can see that the orange area (which IMHO should be red, because whilst I appreciate that my lack of color-blindness is why there's a difference in clarity, still, red is the danger zone in most colour schemes) is set at 6.  Before I had no way of telling where the meter's danger zone is set to, so wouldn't be able to trust it.  I'd be all for making that a preference, but I don't think it should be defailt.

**Comment by Be-ing** [comment]
> LUFS as selectable alternative to RMS

Tenacity has its own homebaked implementation of EBUR128. Why, I have no idea. This should likely be replaced with [libebur128](https://github.com/jiixyj/libebur128).
